### ISSUE TITLE
Acces to weights after quantization

### ISSUE THREAD

**Issue by TanasaAlinConstantin** [pytorch/pytorch#33013]
Hi! 

I have a model quantized with torch.quantization. I saved the model using torch.jit.save and after loading the model it seems that I don't have access to the weights. 

It's not yet fully developed or some other aspects must be taken into consideration when saving with jit a quantized model? 

I use:
torch.jit.save(torch.jit.script(model), "test_model.pkl")

It seems that a full-precision model saved with jit has its weights available through "model.conv1.weights" but not when it's quantized. 

Thank you!

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by RuidongWu** [comment]
Almost you can save the state_dict through "torch.save(model.state_dict,'model_state_dict.pth')".
and then reload this state_dict:
"model_quantized_state_dict = torch.load('model_state_dict.pth')"

If you can print the parameters of model_quantized_state_dict, you can also have the access of each layer. For example:
********************
for name in model_quantized_state_dict:
        print(model_quantized_state_dict[name])
********************
According to the above result, "_weight, bias, scale zero_point_" of layer is shown.
Note: _model_quantized_state_dict['layer0.weight'].int_repr()_ could change the dtype from qint8 to int8.

This is my understanding and suggestion, but I cannot guarantee the accuracy of the results.

**Comment by TanasaAlinConstantin** [comment]
The problem is that you can only save quantized models with jit. So implicitly the load is done with torch.jit.load.
And in this situation, the problem can be seen, not state_dict.

**Comment by TanasaAlinConstantin** [comment]
Thank you very much! 
It works if it is saved with state_dict
